### PR TITLE
Fix siteground caching issue

### PR DESCRIPTION
**Pull request by adi3890**
### Description
- Resolves Site Ground caching issue
[JIRA Card](https://brainstormforce.atlassian.net/browse/SPCTR-3517)

### Screenshots
[https://d.pr/v/x74GZs](https://d.pr/v/x74GZs)

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-caching.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+		if ( ! isset( self::$instance ) || null === self::$instance ) {
+			self::$instance = new self();
+
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor
```
Comment (bsf-zanev) [inline classes/class-uagb-caching.php:44]: We could have a since tag on this class, and this constructor.

File: classes/class-uagb-caching.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	 */
+	public function __construct() {
+		add_action( 'uagb_delete_uag_asset_dir', array( $this, 'clear_cache' ) );
+		add_action( 'uagb_delete_page_assets', array( $this, 'clear_cache' ) );
```
Comment (bsf-zanev) [inline classes/class-uagb-caching.php:48]: Do we have to have these two as separate actions, or can we have a single action`uagb_delete_cached_assets` that is done in `delete_page_assets()` as well as `delete_uag_asset_dir()`?

Comment (adi3890) [reply classes/class-uagb-caching.php:48]: Keeping separate actions gives more flexibility as some other specific action can be performed.
In future if some one want to add any functionality only after delete_page assets that can be manages separately.

File: classes/class-uagb-caching.php (hunk 1 above)
Comment (adi3890) [reply classes/class-uagb-caching.php:46]: done

**Review by bsf-zanev — APPROVED** [review]
Looks good to me.

**Pull request merged**